lib/git-fastclone.rb

```diff
@@ -288,6 +289,7 @@ def with_reference_repo_lock(url, &block)
     ensure
       # Not strictly necessary to do this unlock as an ensure. If ever exception is caught outside this
       # primitive, ensure protection may come in handy.
+      puts "Waiting for another fastclone process to finish"
```

Comment:
I'm not sure this message is accurate. Aren't we yielding the lock, so other pending fastclone processes can start? What is the behavior you're encountering and trying to understand better?